File: solutions_5648941810974720_1/C++/Goodhope/r1_2016_A.cpp
// gj.cpp
//

#include <assert.h>

#include <fstream>
#include <sstream>
#include <stack>
#include <iomanip>
#include <iostream>
#include <string>
#include <vector>
#include <set>
#include <map>
#include <list>
#include <algorithm>
#include <utility>
#include <numeric>

using namespace std;

typedef unsigned char uchar;
typedef unsigned int uint;
typedef unsigned __int64 uint64;

#define EPS (1E-10)
#define PI 3.1415926535897932384626433832795

//("ZERO", "ONE", "TWO", "THREE", "FOUR", "FIVE", "SIX", "SEVEN", "EIGHT", "NINE")

uint a[256];

uint v[1000],
     vn;

int main(int argc, char* argv[])
{
	uint64 cases;
    cin >> cases;

    for (uint64 cs = 1; cs <= cases; ++cs)
    {
		string s;
		cin >> s;

		uint sz = s.size();
		vn = 0;
		for (uint i = 0; i < sz; ++i)
			++a[s[i]];

		// 0
		for (; a['Z']; --a['Z'])
		{
			v[vn++] = 0;
			--a['E'];
			--a['R'];
			--a['O'];
		}

		// 2
		for (; a['W']; --a['W'])
		{
			v[vn++] = 2;
			--a['T'];
			--a['O'];
		}

		// 6
		for (; a['X']; --a['X'])
		{
			v[vn++] = 6;
			--a['S'];
			--a['I'];
		}

		// 8
		for (; a['G']; --a['G'])
		{
			v[vn++] = 8;
			--a['E'];
			--a['I'];
			--a['H'];
			--a['T'];
		}

		// 3
		for (; a['T']; --a['T'])
		{
			v[vn++] = 3;
			--a['H'];
			--a['R'];
			--a['E'];
			--a['E'];
		}

		// 4
		for (; a['U']; --a['U'])
		{
			v[vn++] = 4;
			--a['F'];
			--a['O'];
			--a['R'];
		}

		// 5
		for (; a['F']; --a['F'])
		{
			v[vn++] = 5;
			--a['I'];
			--a['V'];
			--a['E'];
		}

		// 1
		for (; a['O']; --a['O'])
		{
			v[vn++] = 1;
			--a['N'];
			--a['E'];
		}

		// 7
		for (; a['V']; --a['V'])
		{
			v[vn++] = 7;
			--a['S'];
			--a['E'];
			--a['E'];
			--a['N'];
		}

		// 9
		for (; a['I']; --a['I'])
		{
			v[vn++] = 9;
			--a['N'];
			--a['N'];
			--a['E'];
		}

		sort(v, v + vn);
		cout << "Case #" << cs << ": ";

		for (uint i = 0; i < vn; ++i)
			cout << v[i];
		cout << '\n';
    }

    return 0;
}

File: solutions_5648941810974720_1/Python/Tedium/R12016A.py
from string import *

f = open('A-large.in', 'r')
inp = f.read().split("\n")
f.close()

inds = {}
for i in range(len(ascii_uppercase)):
    inds[ascii_uppercase[i]] = i

T = int(inp[0])

def update (c, k, num):
    count = c[inds[k]]
    for i in num:
        c[inds[i]] -= count
    return (count, c)

for itr in range(1, T+1):
    s = inp[itr]

    c = [0] * 26
    for i in s:
        c[inds[i]] += 1

    numbs = [0] * 10
    (numbs[0], c) = update(c, 'Z', 'ZERO')
    (numbs[2], c) = update(c, 'W', 'TWO')
    (numbs[6], c) = update(c, 'X', 'SIX')
    (numbs[8], c) = update(c, 'G', 'EIGHT')
    (numbs[4], c) = update(c, 'U', 'FOUR')
    (numbs[3], c) = update(c, 'R', 'THREE')
    (numbs[1], c) = update(c, 'O', 'ONE')
    (numbs[5], c) = update(c, 'F', 'FIVE')
    (numbs[7], c) = update(c, 'V', 'SEVEN')
    (numbs[9], c) = update(c, 'I', 'NINE')

    out = ""
    for i in range(10):
        out += str(i) * numbs[i]
    print("Case #%d: %s" % (itr, out))
